_posts/2016-09-06-bazel-rules-protobuf.md

```diff
@@ -0,0 +1,1008 @@
+17:33:10 1001 hellos (0 errs, 7998 remaining): 9.0 hellos/ms or ~111µs per hello
+17:33:10 1001 hellos (0 errs, 6997 remaining): 9.1 hellos/ms or ~110µs per hello
+17:33:10 1001 hellos (0 errs, 5996 remaining): 8.6 hellos/ms or ~116µs per hello
+17:33:10 1001 hellos (0 errs, 4995 remaining): 9.0 hellos/ms or ~111µs per hello
+17:33:10 1001 hellos (0 errs, 3994 remaining): 9.0 hellos/ms or ~111µs per hello
+17:33:10 1001 hellos (0 errs, 2993 remaining): 9.1 hellos/ms or ~110µs per hello
+17:33:10 1001 hellos (0 errs, 1992 remaining): 9.0 hellos/ms or ~111µs per hello
+17:33:10 1001 hellos (0 errs, 991 remaining): 9.0 hellos/ms or ~111µs per hello
+17:33:11 991 hellos (0 errs, -1 remaining): 9.0 hellos/ms or ~111µs per hello
+```
+
+```sh
+# Timings for the C# server
+~/rules_protobuf$ bazel run examples/helloworld/csharp/GreeterServer
+
+~/grpc_greeterclient$ bazel run //go:client -- -total_size 10000 -batch_size 1000
+17:34:37 1001 hellos (0 errs, 8999 remaining): 6.0 hellos/ms or ~166µs per hello
+17:34:37 1001 hellos (0 errs, 7998 remaining): 6.7 hellos/ms or ~150µs per hello
+17:34:37 1001 hellos (0 errs, 6997 remaining): 6.8 hellos/ms or ~148µs per hello
+17:34:37 1001 hellos (0 errs, 5996 remaining): 6.8 hellos/ms or ~147µs per hello
+17:34:37 1001 hellos (0 errs, 4995 remaining): 6.7 hellos/ms or ~150µs per hello
+17:34:38 1001 hellos (0 errs, 3994 remaining): 6.7 hellos/ms or ~150µs per hello
+17:34:38 1001 hellos (0 errs, 2993 remaining): 6.7 hellos/ms or ~149µs per hello
+17:34:38 1001 hellos (0 errs, 1992 remaining): 6.7 hellos/ms or ~149µs per hello
+17:34:38 1001 hellos (0 errs, 991 remaining): 6.8 hellos/ms or ~148µs per hello
+17:34:38 991 hellos (0 errs, -1 remaining): 6.8 hellos/ms or ~147µs per hello
+```
+
+The informal analysis demonstrated comparable timings for c++, go, and
+java greeter service implementations.  The c++ server had the overall
+fastest and most consistent performance.  The go implementation was
+also very consistent, but slightly slower than C++.  Java demonstrated
+some initial relative slowness likely due to the JVM warming up but
+soon converged on timings similar to the C++ implementation.  C# has
+consistent performance but marginally slower.
+
+## 2.5: Summary
+
+Bazel assists in the construction of gRPC applications by providing a
+capable build environment for services built in a multitude of
+languages.  rules_protobuf complements bazel by packaging up all the
+dependencies needed and abstracting away the need to call protoc
+directly.
+
+In this workflow one does need to check in the generated source code
+as it is always generated on-demand within your workspace.  For
+projects that do need this, one can manually copy over the generated
+files or use the `output_to_workspace` option to place the generated
+files alongside the protobuf definitions.
+
+In addition to helping with c++, java, and golang code, rules_protobuf
+also supports (in part) javascript, ruby, and python, but not yet with
```

Comment:
I had to read that sentence a few times to understand that the "not yet with gRPC" part only applies to Javascript, Ruby and Python :-)